src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3542,24 +3566,85 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             int n = arguments.Length;
-            if (n == 0)
+            if (n == 0 && parametersOpt.IsDefaultOrEmpty)
             {
-                return ImmutableArray<VisitArgumentResult>.Empty;
+                return (ImmutableArray<VisitArgumentResult>.Empty, argsToParamsOpt, refKindsOpt);
             }
-            var builder = ArrayBuilder<VisitArgumentResult>.GetInstance(n);
+
+            var visitedParameters = PooledHashSet<ParameterSymbol>.GetInstance();
+            var resultsBuilder = ArrayBuilder<VisitArgumentResult>.GetInstance(n);
             for (int i = 0; i < n; i++)
             {
-                FlowAnalysisAnnotations parameterAnnotations = GetCorrespondingParameter(i, parameters, argsToParamsOpt, expanded).Annotations;
-                builder.Add(VisitArgumentEvaluate(arguments[i], GetRefKind(refKindsOpt, i), parameterAnnotations));
+                var (parameter, _, parameterAnnotations, _) = GetCorrespondingParameter(i, parametersOpt, argsToParamsOpt, expanded);
+
+                resultsBuilder.Add(VisitArgumentEvaluate(arguments[i], GetRefKind(refKindsOpt, i), parameterAnnotations));
+                visitedParameters.Add(parameter);
+            }
+
+            if (!parametersOpt.IsDefaultOrEmpty)
+            {
+                ArrayBuilder<int>? argsToParamsBuilder;
+                if (argsToParamsOpt.IsDefault)
+                {
+                    argsToParamsBuilder = null;
+                }
+                else
+                {
+                    argsToParamsBuilder = ArrayBuilder<int>.GetInstance(parametersOpt.Length);
+                    argsToParamsBuilder.AddRange(argsToParamsOpt);
+                }
+
+                ArrayBuilder<RefKind>? argRefKindsBuilder;
+                if (refKindsOpt.IsDefault)
+                {
+                    argRefKindsBuilder = null;
+                }
+                else
+                {
+                    argRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(parametersOpt.Length);
+                    argRefKindsBuilder.AddRange(refKindsOpt);
+                }
+
+                var previousDisableNullabilityAnalysis = _disableNullabilityAnalysis;
+                _disableNullabilityAnalysis = true;
+                for (int i = 0; i < parametersOpt.Length; i++)
+                {
+                    var parameter = parametersOpt[i];
+                    // fill in unspecified optional arguments
+                    if (parameter.IsOptional && !visitedParameters.Contains(parameter))
+                    {
+                        var annotations = GetParameterAnnotations(parameter);
+                        if (!_defaultValues.TryGetValue((syntax, parameter), out var argument))
+                        {
+                            _defaultValues[(syntax, parameter)] = argument = LocalRewriter.GetDefaultParameterValue(syntax, parameter, ThreeState.Unknown, localRewriter: null, _binder, Diagnostics);
```

Comment:
>Unknown [](start = 146, length = 7)

Should this be `ThreeState.True` instead?

Reply:
I think so, because we know we are in an invocation. It seems that passing Unknown makes it examine the syntax to decide if the default argument is contained in a kind of syntax where caller info makes sense.

---
In reply to: [329256664](https://github.com/dotnet/roslyn/pull/38788#discussion_r329256664) [](ancestors = 329256664)